lib/jxl/base/data_parallel.h

```diff
@@ -104,21 +97,33 @@ class ThreadPool {
   void* const runner_opaque_;
 };
 
-// TODO(deymo): Convert the return value to a Status when not using SkipInit.
 template <class InitFunc, class DataFunc>
-bool RunOnPool(ThreadPool* pool, const uint32_t begin, const uint32_t end,
-               const InitFunc& init_func, const DataFunc& data_func,
-               const char* caller) {
-  Status ret = true;
+Status RunOnPool(ThreadPool* pool, const uint32_t begin, const uint32_t end,
+                 const InitFunc& init_func, const DataFunc& data_func,
+                 const char* caller) {
+  if (pool == nullptr) {
+    ThreadPool default_pool(nullptr, nullptr);
+    return default_pool.Run(begin, end, init_func, data_func, caller);
+  } else {
+    return pool->Run(begin, end, init_func, data_func, caller);
+  }
+}
+template <class DataFunc>
+Status RunOnPoolSkipInit(ThreadPool* pool, const uint32_t begin,
+                         const uint32_t end, const DataFunc& data_func,
+                         const char* caller) {
```

Comment:
I believe now we can just have this call RunOnPool without duplicating code :)

(although perhaps at this point I would remove this function entirely...)

Reply:
Right. Removed the function.